### PR TITLE
WIP:add new random points generating def

### PR DESCRIPTION
**Pull request by justinkotzebue**
generate_random_points needed some adjustment in order to work under rasterio v. 1.09a10

Can we delete the function which only works with the old version of rasterio?

### CODE REVIEW

File: gis_utils/r/sample_values.py [hunk 1]
```diff
@@ -37,6 +37,7 @@ def return_lat_lon(gt, x, y, gt_type="rasterio"):
 
 def _generate_random_coordinates(r, num_points):
```
Comment (j08lue) [inline gis_utils/r/sample_values.py]: Just remove this if we do not need it anymore.

File: gis_utils/r/sample_values.py [hunk 2]
```diff
@@ -68,6 +69,35 @@ def _generate_random_coordinates(r, num_points):
         return [center(to_xy(((rand(h), rand(w))))) for i in range(num_points)]
 
 
+def generate_random_coordinates(r, num_points):
+    """
```
Comment (j08lue) [inline gis_utils/r/sample_values.py]: Please reformat the docstring a bit: The first line has to be the short description of what the function does. And there need to be blank lines before headings like `Parameters`. Links to credits go under some `Credits` or `Sources` heading. And please leave out stuff like dependency specs. I have never seen those in function docstrings.

File: gis_utils/r/sample_values.py [hunk 3]
```diff
@@ -68,6 +69,35 @@ def _generate_random_coordinates(r, num_points):
...
+        The X,Y coordinates of random pixel centers
+    """
+    # Get dimensions and cellsize of raster
+    w, h = r.meta['width'], r.meta['height']
+    # Create the functions to generate the random points
+    rand = lambda lim: np.random.randint(lim)
+
+    # Generate the list of points
+    if int(sys.version[0]) < 3:
+        return [r.xy(rand(h), rand(w)) for i in xrange(num_points)]
```
Comment (j08lue) [inline gis_utils/r/sample_values.py]: `range(num_points` works in Python 2 as well, just not as efficiently. Bad luck for those still using Python 2.

Unless really necessary, write code that works with Python 2 and 3 without `sys.version` checks.

File: gis_utils/r/sample_values.py [hunk 4]
```diff
@@ -68,6 +69,35 @@ def _generate_random_coordinates(r, num_points):
...
+        The number of points to generate
+    Returns
+    -------
+    points: list of tuples
+        The X,Y coordinates of random pixel centers
+    """
+    # Get dimensions and cellsize of raster
+    w, h = r.meta['width'], r.meta['height']
+    # Create the functions to generate the random points
+    rand = lambda lim: np.random.randint(lim)
```
Comment (j08lue) [inline gis_utils/r/sample_values.py]: This lambda does nothing but make a wrapper for `np.random.randint`. A function that calls a function and passes on the single argument it gets. That gives you function call overhead and obscurity, both of which you do not want. :smile:

Comment (justinkotzebue) [reply gis_utils/r/sample_values.py]: haha okay, did not look too much into the code which I just found somewhere :)

File: gis_utils/r/sample_values.py [hunk 5]
```diff
@@ -47,54 +47,75 @@ def _generate_random_coordinates(r, num_points):
...
     # Generate the list of points
-    if int(sys.version[0]) < 3:
-        return [center(to_xy(((rand(h), rand(w))))) for i in xrange(num_points)]
-    else:
-        return [center(to_xy(((rand(h), rand(w))))) for i in range(num_points)]
+    return [r.xy(rand(h), rand(w)) for i in range(num_points)]
 
 
-def extract_values(open_rasterio, array_to_extract, number_of_coordinates):
+def extract_values(open_rasterio, array_to_extract, input_points=None, random_coordinates=None):
```
Comment (j08lue) [inline gis_utils/r/sample_values.py]: What here is called `open_rasterio` I see usually referenced as `src` or `source`. Maybe we should make that a convention?

https://github.com/mapbox/rasterio/blob/7602fd6aadd6edb80e2b82d6b9bacbbab81a9a8b/rasterio/merge.py#L18

Comment (justinkotzebue) [reply gis_utils/r/sample_values.py]: yes you're right, maybe lets use `src` instead of `source` since that is how rasterio uses it?

Comment (j08lue) [reply gis_utils/r/sample_values.py]: `src` is for internal use, like `with rasterio.open(path) as src`, but I think `source` is better for a public interface and that is how `rasterio` does it in `merge.merge` or `warp.reproject`:

https://github.com/mapbox/rasterio/blob/7602fd6aadd6edb80e2b82d6b9bacbbab81a9a8b/rasterio/warp.py#L176

**Pull request merged**